### PR TITLE
Ensure Discovery Queries only return indexed fields

### PR DESCRIPTION
**Pull request by foster33**
Fixes the problem where Discovery Queries could return fields, that were not marked as indexed in the metadata table, as results.

Also adds a unit test to validate the fix.

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java [hunk 1]
```diff
@@ -379,7 +384,12 @@ public Iterator<DiscoveredThing> apply(Entry<Key,Value> from) {
                 }
                 ArrayList<DiscoveredThing> thangs = Lists.newArrayListWithCapacity(aw.get().length);
                 for (Writable w : aw.get()) {
-                    thangs.add((DiscoveredThing) w);
+                    // Check to see if the field is currently indexed, if it's not, we should NOT be adding it to 'thangs'
+                    if (indexedFields.contains(((DiscoveredThing) w).getField())) {
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java:648]: opportunity to rename existing variable to something more adult sounding like "items"?

File: warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java [hunk 2]
```diff
@@ -75,6 +75,7 @@ public class DiscoveryLogic extends ShardIndexQueryTable {
     public static final String SEPARATE_COUNTS_BY_COLVIS = "separate.counts.by.colvis";
     public static final String SHOW_REFERENCE_COUNT = "show.reference.count";
     public static final String REVERSE_INDEX = "reverse.index";
+    private static Set<String> indexedFields;
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java]: I wonder if this needs to be static.

Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java]: +1

File: warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java (hunk 1 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java:648]: I protest, we are not adults here

**Review by ivakegg — CHANGES_REQUESTED** [review]
Except for the static member, I am good with this

File: warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java (hunk 2 above)
Comment (foster33) [reply warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java]: Fixed in latest commit

**Review by apmoriarty — DISMISSED** [review]

**Review by ivakegg — DISMISSED** [review]

**Review by apmoriarty — APPROVED** [review]

**Review by ivakegg — APPROVED** [review]

**Pull request merged**